Doc build fails on Windows

The first error is not shown, if the `build` folder is created manually.

```cmd
make[1]: Entering directory `C:/dev/kivy/doc'
mkdir build\\html build\\doctrees
mkdir: cannot create directory `build\\html': No such file or directory
mkdir: cannot create directory `build\\doctrees': No such file or directory
make[1]: *** [html] Error 1
make[1]: Leaving directory `C:/dev/kivy/doc'
make: *** [html] Error 2
```

And at the end of the build:
```cmd
copying static files... done
copying extra files... done
dumping search index... done
dumping object inventory... done
build succeeded.
sh: C:\Program: No such file or directory
make[1]: *** [html] Error 127
make[1]: Leaving directory `C:/dev/kivy/doc'
make: *** [html] Error 2
```

<bountysource-plugin>
---
Want to back this issue? **[Post a bounty on it!](https://www.bountysource.com/issues/5410566-doc-build-fails-on-windows?utm_campaign=plugin&utm_content=tracker%2F42681&utm_medium=issues&utm_source=github)** We accept bounties via [Bountysource](https://www.bountysource.com/?utm_campaign=plugin&utm_content=tracker%2F42681&utm_medium=issues&utm_source=github).
</bountysource-plugin>

I've just checked again with the new stack, i'm getting a new error:
```
$ make html
python setup.py build_ext --inplace
Using distutils
Environ change use_sdl2 -> True
Environ change use_gstreamer -> False

Detected Cython version 0.24
  This version of Cython is untested with Kivy. While this version may
  work perfectly fine, it is possible that you may experience issues. If
  you do have issues, please downgrade to a supported version. It is
  best to use the newest supported version, 0.23, but the minimum
  supported version is 0.20.

  If your platform provides a Cython package, check if you can downgrade
  to a supported version. Otherwise, uninstall the platform package and
  install Cython via pip:

    pip install -I Cython==0.23
User distribution detected, avoid portable command.
Windows platform detected, force GLEW usage.
Using this graphics system: OpenGL
WARNING: A problem occured while running pkg-config --libs --cflags sdl2 SDL2_ttf SDL2_image SDL2_mixer (code 1)

b"'pkg-config' is not recognized as an internal or external command,\r\noperable program or batch file.\r\n"

SDL2: found SDL header at C:\Python34\include\SDL2\SDL.h
SDL2: found SDL_mixer header at C:\Python34\include\SDL2\SDL_mixer.h
SDL2: found SDL_ttf header at C:\Python34\include\SDL2\SDL_ttf.h
SDL2: found SDL_image header at C:\Python34\include\SDL2\SDL_image.h
 C:\Python34\lib\distutils\dist.py:260: UserWarning: Unknown distribution option: 'setup_requires'
   warnings.warn(msg)
 C:\Python34\lib\distutils\dist.py:260: UserWarning: Unknown distribution option: 'install_requires'
   warnings.warn(msg)
 C:\Python34\lib\distutils\dist.py:260: UserWarning: Unknown distribution option: 'dependency_links'
   warnings.warn(msg)
running build_ext
Build configuration is:
 * use_rpi = 0
 * use_mali = 0
 * use_egl = 0
 * use_opengl_es2 = 0
 * use_opengl_debug = 0
 * use_opengl_mock = 0
 * use_glew = 1
 * use_sdl2 = 1
 * use_ios = 0
 * use_mesagl = 0
 * use_x11 = 0
 * use_gstreamer = 0
 * use_avfoundation = 0
 * use_osx_frameworks = 0
 * debug_gl = 0
 * debug = False
Detected compiler is mingw32
skipping 'kivy\core/audio\audio_sdl2.c' Cython extension (up-to-date)
skipping 'kivy\core/text\text_layout.c' Cython extension (up-to-date)
skipping 'kivy\graphics\texture.c' Cython extension (up-to-date)
skipping 'kivy\graphics\context_instructions.c' Cython extension (up-to-date)
skipping 'kivy\graphics\stencil_instructions.c' Cython extension (up-to-date)
skipping 'kivy\graphics\context.c' Cython extension (up-to-date)
skipping 'kivy\_event.c' Cython extension (up-to-date)
skipping 'kivy\graphics\vertex.c' Cython extension (up-to-date)
skipping 'kivy\graphics\svg.c' Cython extension (up-to-date)
skipping 'kivy\graphics\transformation.c' Cython extension (up-to-date)
skipping 'kivy\core/text\_text_sdl2.c' Cython extension (up-to-date)
skipping 'kivy\graphics\c_opengl_debug.c' Cython extension (up-to-date)
skipping 'kivy\graphics\opengl.c' Cython extension (up-to-date)
skipping 'kivy\graphics\compiler.c' Cython extension (up-to-date)
skipping 'kivy\graphics\vbo.c' Cython extension (up-to-date)
skipping 'kivy\graphics\buffer.c' Cython extension (up-to-date)
skipping 'kivy\graphics\fbo.c' Cython extension (up-to-date)
skipping 'kivy\graphics\gl_instructions.c' Cython extension (up-to-date)
skipping 'kivy\graphics\instructions.c' Cython extension (up-to-date)
skipping 'kivy\weakproxy.c' Cython extension (up-to-date)
skipping 'kivy\graphics\opengl_utils.c' Cython extension (up-to-date)
skipping 'kivy\graphics\shader.c' Cython extension (up-to-date)
skipping 'kivy\graphics\tesselator.c' Cython extension (up-to-date)
skipping 'kivy\core/image\_img_sdl2.c' Cython extension (up-to-date)
skipping 'kivy\core/window\_window_sdl2.c' Cython extension (up-to-date)
skipping 'kivy\graphics\scissor_instructions.c' Cython extension (up-to-date)
skipping 'kivy\properties.c' Cython extension (up-to-date)
skipping 'kivy\core/clipboard\_clipboard_sdl2.c' Cython extension (up-to-date)
skipping 'kivy\graphics\c_opengl_mock.c' Cython extension (up-to-date)
skipping 'kivy\graphics\vertex_instructions.c' Cython extension (up-to-date)
cd doc && C:/Python34/share/mingwpy/bin/mingw32-make html
mingw32-make[1]: Entering directory 'E:/dev/prs/kivy/doc'
mkdir -p build/html build/doctrees
python autobuild.py silenced=yes
[INFO              ] [Logger      ] Record log in C:\Users\prc\.kivy\logs\kivy_16-05-07_0.txt
[INFO              ] [Kivy        ] v1.9.2-dev0
[INFO              ] [Python      ] v3.4.4 (v3.4.4:737efcadf5a6, Dec 20 2015, 19:28:18) [MSC v.1600 32 bit (Intel)]
[INFO              ] [Factory     ] 193 symbols loaded
[INFO              ] [OSC         ] using <thread> for socket
[ERROR             ] [Input       ] AndroidJoystick is not supported by your version of linux
Traceback (most recent call last):
  File "E:\dev\prs\kivy\kivy\input\providers\__init__.py", line 57, in <module>
    import kivy.input.providers.androidjoystick
  File "E:\dev\prs\kivy\kivy\input\providers\androidjoystick.py", line 27, in <module>
    import pygame.joystick
ImportError: No module named 'pygame'
 Traceback (most recent call last):
   File "autobuild.py", line 64, in <module>
     from kivy.lib import osc, ddsfile, mtdev
   File "E:\dev\prs\kivy\kivy\lib\mtdev.py", line 27, in <module>
     libmtdev = cdll.LoadLibrary('libmtdev.so.1')
   File "C:\Python34\lib\ctypes\__init__.py", line 429, in LoadLibrary
     return self._dlltype(name)
   File "C:\Python34\lib\ctypes\__init__.py", line 351, in __init__
     self._handle = _dlopen(self._name, mode)
 OSError: [WinError 126] The specified module could not be found
makefile:80: recipe for target 'html' failed
mingw32-make[1]: *** [html] Error 1
mingw32-make[1]: Leaving directory 'E:/dev/prs/kivy/doc'
makefile:71: recipe for target 'html' failed
mingw32-make: *** [html] Error 2
```

When trying to remove dll import from mtdev.py, nothing changes, still the same error raised even if that line is not present in the file. However if commented like this: `from kivy.lib import osc, ddsfile#, mtdev` I get new error(what is the error actually? O.o). Also, the solution isn't good, because you still need that file to be included in documentation.

<pre>[DEBUG             ] [Cache       ] register <textinput.width> with limit=None,
timeout=60.0
[INFO              ] [OSC         ] using <thread> for socket
[ERROR             ] [Input       ] AndroidJoystick is not supported by your ver
sion of linux
Traceback (most recent call last):
  File "C:\Users\John\Desktop\python\lib\site-packages\kivy\input\providers\__in
it__.py", line 57, in <module>
    import kivy.input.providers.androidjoystick
  File "C:\Users\John\Desktop\python\lib\site-packages\kivy\input\providers\andr
oidjoystick.py", line 27, in <module>
    import pygame.joystick
ImportError: No module named pygame.joystick
[CRITICAL          ] [App         ] Unable to get a Window, abort.
 Exception SystemExit: 1 in 'kivy.properties.dpi2px' ignored
[DEBUG             ] [Cache       ] register <kv.loader> with limit=500, timeout
=60
[CRITICAL          ] [App         ] Unable to get a Window, abort.
 Exception SystemExit: 1 in 'kivy.properties.dpi2px' ignored
process_begin: CreateProcess(NULL, touch autobuild.py-done-with-python, ...) fai
led.
make (e=2): The system cannot find the file specified.
mingw32-make: *** [html] Error 2</pre>

in Makefile I put a space `autobuild.py<here>-done-with-`(seems as a weird command to me anyway when used without space) and result

<pre>mingw32-make: [clean] Error 2 (ignored)
rmdir /s /q build
[INFO              ] [Logger      ] Record log in C:\Users\John\.kivy\logs\kivy_
16-05-07_42.txt
[INFO              ] [Kivy        ] v1.9.1
[INFO              ] [Python      ] v2.7.11 (v2.7.11:6d1b6a68f775, Dec  5 2015,
20:32:19) [MSC v.1500 32 bit (Intel)]
[INFO              ] [gallery.py  ] Created gallery rst documentation pages.
process_begin: CreateProcess(NULL, sphinx-build -b html -d build/doctrees -W -T
sources build/html, ...) failed.
make (e=2): The system cannot find the file specified.
mingw32-make: *** [html] Error 2</pre>

--- a/doc/Makefile
+++ b/doc/Makefile
@@ -40,7 +40,7 @@ else
 endif
 
 AUTOBUILD_STAMP_BASE = autobuild.py-done-with-
-AUTOBUILD_STAMP      = $(AUTOBUILD_STAMP_BASE)$(PYTHON)
+AUTOBUILD_STAMP      = "$(AUTOBUILD_STAMP_BASE)$(PYTHON)"
 
 .PHONY: help clean html web pickle htmlhelp latex changes linkcheck gettext
 
@@ -80,7 +80,7 @@ html:
 	$(MKDIR) build$(P)html build$(P)doctrees
 ifeq ("$(wildcard $(AUTOBUILD_STAMP))","")
 	$(PYTHON) autobuild.py silenced=$(ENDUSER_BUILD)
-	touch $(AUTOBUILD_STAMP)
+	touch $(AUTOBUILD_STAMP) || type NUL > $(AUTOBUILD_STAMP)
 endif
 	$(PYTHON) gallery.py
 	$(SPHINXBUILD) -b html $(ALLSPHINXOPTS$(_TESTS)) build/html

--- a/kivy/lib/mtdev.py
+++ b/kivy/lib/mtdev.py
@@ -24,7 +24,8 @@ from ctypes import cdll, Structure, c_ulong, c_int, c_ushort, \
                    c_void_p, pointer, POINTER, byref
 
 # load library
-libmtdev = cdll.LoadLibrary('libmtdev.so.1')
+if 'KIVY_DOC' not in os.environ:
+    libmtdev = cdll.LoadLibrary('libmtdev.so.1')
 
 # from linux/input.h
 MTDEV_CODE_SLOT          = 0x2f  # MT slot being modified
@@ -122,14 +123,15 @@ class mtdev(Structure):
     ]
 
 # binding
-mtdev_open = libmtdev.mtdev_open
-mtdev_open.argtypes = [POINTER(mtdev), c_int]
-mtdev_get = libmtdev.mtdev_get
-mtdev_get.argtypes = [POINTER(mtdev), c_int, POINTER(input_event), c_int]
-mtdev_idle = libmtdev.mtdev_idle
-mtdev_idle.argtypes = [POINTER(mtdev), c_int, c_int]
-mtdev_close = libmtdev.mtdev_close
-mtdev_close.argtypes = [POINTER(mtdev)]
+if 'KIVY_DOC' not in os.environ:
+    mtdev_open = libmtdev.mtdev_open
+    mtdev_open.argtypes = [POINTER(mtdev), c_int]
+    mtdev_get = libmtdev.mtdev_get
+    mtdev_get.argtypes = [POINTER(mtdev), c_int, POINTER(input_event), c_int]
+    mtdev_idle = libmtdev.mtdev_idle
+    mtdev_idle.argtypes = [POINTER(mtdev), c_int, c_int]
+    mtdev_close = libmtdev.mtdev_close
+    mtdev_close.argtypes = [POINTER(mtdev)]
 
 
 class Device:

